src/_pytest/fixtures.py

```diff
@@ -437,14 +442,18 @@ def __init__(self, pyfuncitem, *, _ispytest: bool = False) -> None:
         self._pyfuncitem = pyfuncitem
         #: Fixture for which this request is being performed.
         self.fixturename: Optional[str] = None
-        #: Scope string, one of "function", "class", "module", "session".
-        self.scope: _Scope = "function"
+        self._scope = Scope.Function
         self._fixture_defs: Dict[str, FixtureDef[Any]] = {}
         fixtureinfo: FuncFixtureInfo = pyfuncitem._fixtureinfo
         self._arg2fixturedefs = fixtureinfo.name2fixturedefs.copy()
         self._arg2index: Dict[str, int] = {}
         self._fixturemanager: FixtureManager = pyfuncitem.session._fixturemanager
 
+    @property
+    def scope(self) -> "_ScopeName":
+        """Backward-compatibility to return the scope string value."""
```

Comment:
As long as we're not exposing `Scope` and `_scope` is private, I don't think we should document it as backward-compat, since we don't expect anyone to use something else.

Reply:
Agreed. But on second thought perhaps we can ditch the property completely? I find it a bit confusing to have `scope: str` and `_scope: Scope`. If we don't expect to support backward compatibility here, given it is an internal object, might make sense to just drop the property entirely.

Reply:
I don't think it's an internal object. I'll be surprised if no one is using `request.scope`. It is also [documented](https://docs.pytest.org/en/latest/reference/reference.html#pytest.FixtureRequest.scope).

One option is to use an `Enum` which also derives from `str`, then it will be compatible. From my experience, these cause a lot of confusion, but may make sense here.

Reply:
Hmm then I think the backward compatibility bit on the docstring makes sense TBH, we have that difference only because it is part of the public API.

Reply:
What I mean to say is that we shouldn't designate something as backward compatibility when it doesn't have a replacement. The new attribute is not a replacement because it is private.

Reply:
Hmm not sure I follow, but I don't feel strongly about it either. Can you tell what docstring exactly you would like to see there then? I'm happy to change it.

Reply:
> Hmm not sure I follow

It's probably me. Maybe your intention is to *remove* `request.scope` without replacement? If so, I'm not sure about it, I think it's useful.

> Can you tell what docstring exactly you would like to see there then

Can use the existing docstring

> Scope string, one of "function", "class", "module", "session".

although it's missing "package" so should add that.